dimod/binary_quadratic_model.py

```diff
@@ -1286,23 +1311,23 @@ def binary_to_spin(linear, quadratic, offset):
         """
         h = {}
         J = {}
-        linear_offset = 0.0
-        quadratic_offset = 0.0
+        linear_offset = 0
+        quadratic_offset = 0
 
         for u, bias in iteritems(linear):
-            h[u] = .5 * bias
+            h[u] = bias / 2
             linear_offset += bias
 
         for (u, v), bias in iteritems(quadratic):
 
-            J[(u, v)] = .25 * bias
+            J[(u, v)] = bias / 4
```

Comment:
This will allow for some strange behavior, for example:

```python
>>> qubo_bqm = dimod.BinaryQuadraticModel({}, {(0, 1): 3}, 0, vartype=dimod.BINARY, dtype=np.int64)
>>> ising_bqm = qubo_bqm.change_vartype(dimod.SPIN)
>>> ising_bqm
BinaryQuadraticModel({0: 0, 1: 0}, {(0, 1): 0}, 0.75, Vartype.SPIN)
```

Note that the coupler bias has been erased due to calling `np.int64(0.75)`. Perhaps more importantly,
```python
>>> type(ising_bqm.offset)
numpy.float64
```
Looks like `add_offset` needs to do the same coercion to the dtype that the other `add` methods do.